core/ledger/kvledger/txmgmt/statedb/statecouchdb/cache.go

```diff
@@ -88,7 +89,16 @@ func (c *cache) putState(chainID, namespace, key string, cacheValue *CacheValue)
 	if err != nil {
 		return err
 	}
-	cache.Set(cacheKey, valBytes)
+	// If payload is bigger than fastcache can store then delete previous key.
+	// When GetState() has been called for deleted key then it will call database
+	// against fastcache.
+	if cap(valBytes) > fastCacheValueSizeLimit {
```

Comment:
- `len(valBytes)` instead of `cap(valBytes)`

- Looking closely, unfortunately, even this is not going to be a sufficient condition because they way this library encodes the data internally, `uint64(len(kvLenBuf) + len(k) + len(v))` is the length that matters not the length of the value only. Here, again `kvLenBuf` is a fixed size of `4`. But I do not prefer to depend on so much of internal logic of a library. So, my suggestion would be to always delete the key irrespective of the size condition.

Reply:
> So, my suggestion would be to always delete the key irrespective of the size condition.

@przemyslaw A clarification on the above suggestion. I think @manish-sethi suggested to remove `fastCacheValueSizeLimit` altogether and all related checks associated with it. We will simply delete the existing entry before storing a new one without checking the size. Hence, before every `cache.Set()`, we can simply do `cache.Del()` (while `cache.Has()` is optional). For bigger values, `cache.Set()` would anyway ignore. I have verified this with @manish-sethi as I had some confusion. I am sharing it here so that the suggestion is explicit.

Reply:
+1